### PR TITLE
add vault_config and add zenodo to file sources and user prefs

### PR DESCRIPTION
**Pull request by cat-bro**
to support zenodo integration described in the release notes [here](https://docs.galaxyproject.org/en/master/releases/24.1_announce_user.html#zenodo-integration)

GA has never used the vault feature before so this will need to be tested on dev.

### CODE REVIEW

File: files/galaxy/config/user_preferences_extra_conf.yml [hunk 1]
```diff
@@ -38,6 +29,47 @@ preferences:
...
               type:  password
               required: False
+
+    zenodo:
+        description: Your Zenodo Integration Settings
+        inputs:
+            - name: token
+              label: Personal Access Token used to create draft records and to upload files. You can manage your tokens at https://zenodo.org/account/settings/applications/
+              type: secret
+              # store: vault # Requires setting up vault_config_file in your galaxy.yml
```
Comment (nuwang) [inline files/galaxy/config/user_preferences_extra_conf.yml:39]:
```suggestion
              store: vault # Requires setting up vault_config_file in your galaxy.yml
```

File: files/galaxy/config/user_preferences_extra_conf.yml [hunk 2]
```diff
@@ -38,6 +29,47 @@ preferences:
...
+              type: text
+              required: False
+
+    zenodo_sandbox:
+        description: Your Zenodo Sandbox Integration Settings (TESTING ONLY)
+        inputs:
+            - name: token
+              label: Personal Access Token used to create draft records and to upload files. You can manage your tokens at https://sandbox.zenodo.org/account/settings/applications/
+              type: secret
+              # store: vault # Requires setting up vault_config_file in your galaxy.yml
```
Comment (nuwang) [inline files/galaxy/config/user_preferences_extra_conf.yml:52]:
```suggestion
              store: vault # Requires setting up vault_config_file in your galaxy.yml
```

**Review by nuwang — COMMENTED** [review]
Missing vault_conf.yml.j2?

**Comment by cat-bro** [comment]
Thanks @nuwang. Please do not merge because I'll try this branch on dev first.

**Pull request merged**